packages/medusa-payment-adyen/src/services/googlepay-adyen.js

```diff
@@ -26,12 +28,22 @@ class GooglePayAdyenService extends PaymentService {
     return status
   }
 
-  async createPayment(_) {
-    return {}
+  async createPayment(cart) {
+    const shippingOptions = await this.shippingProfileService_.fetchCartOptions(
+      cart
+    )
+    return {
+      shipping_options: shippingOptions,
+    }
   }
 
-  async authorizePayment(cart, paymentMethod) {
-    return this.adyenService_.authorizePayment(cart, paymentMethod)
+  async authorizePayment(cart, paymentMethod, amount, shopperIp) {
```

Comment:
We should change `authorizePayment` to take a `context` argument in the place of `shopperIp`. The context argument should be an object that can hold additional details about the operation such as the IP address of the client's device.